File: java/src/main/java/nl/craftsmen/workshops/reactivemeetup/domain/Triangle.java
package nl.craftsmen.workshops.reactivemeetup.domain;

public class Triangle implements IShape {

    private final String name;
    private final int base;
    private final int height;

    public Triangle(String name, int base, int height) {
        this.name = name;
        this.base = base;
        this.height = height;
    }

    @Override
    public double calculateArea() {
        return base * height / 2;
    }

    @Override
    public double calculateCircumference() {
        return base + height + Math.sqrt(base * base + height * height);
    }

    @Override
    public String getName() { return name; }
}

File: javascript/src/domain/geometry/triangle.js
module.exports = class Circle {

    constructor(name, base, height) {
        this.name = name;
        this.base = base;
        this.height = height;
    }

    calculateArea() {
        return this.base * this.height / 2;
    }

    calculateCircumference() {
        return this.base + this.height + Math.sqrt(this.base * this.base + this.height * this.height);
    }

    getName() {
        return this.name;
    }
}
